lib/yoti/dynamic_share_service/policy/source_constraint.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module Yoti
+  module DynamicSharingService
+    # A list of anchors to require for a dynamic share
+    class SourceConstraint
+      DRIVING_LICENCE = 'DRIVING_LICENCE'
+      PASSPORT = 'PASSPORT'
+      NATIONAL_ID = 'NATIONAL_ID'
+      PASS_CARD = 'PASS_CARD'
+
+      SOURCE_CONSTRAINT = 'SOURCE'
+
+      attr_reader :anchors
+
+      def soft_preference
+        return @soft_preference if @soft_preference
+
+        false
+      end
+
+      def to_json(*_args)
+        as_json.to_json
+      end
+
+      def as_json
+        obj = {
+          type: SOURCE_CONSTRAINT,
+          preferred_sources: [
+            anchors: @anchors.as_json
+          ]
+        }
+        obj[:preferred_sources][:soft_preference] = @soft_preference if @soft_preference
```

Comment:
Wondering if `soft_preference` should default to `false` like other SDKs? Maybe something to standardise on @echarrod ?

Reply:
I think we've previously said that we should generate minimal json?

Reply:
Setting `.with_soft_preference(false)`, means `soft_preference` wont be included in the payload - it should be present and set to `false` in this situation.

Reply:
Yes, if `.with_soft_preference` is called, we should always add it to the payload 👍